unstructured_ingest/v2/processes/connectors/discord.py

```diff
@@ -0,0 +1,169 @@
+import datetime as dt
+from dataclasses import dataclass
+from typing import Any, Generator, List, Optional, Set
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.logger import logger
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+CONNECTOR_TYPE = "discord"
+
+
+class DiscordAccessConfig(AccessConfig):
+    token: str = Field(description="Discord API token")
+
+
+class DiscordConnectionConfig(ConnectionConfig):
+    access_config: Secret[DiscordAccessConfig]
+    channels: Optional[List[str]] = Field(
```

Comment:
`list` is now a native type in python
```suggestion
    channels: Optional[list[str]] = Field(
```